google/cloud/storage/blob.py

```diff
@@ -190,6 +190,14 @@ def __init__(
         if generation is not None:
             self._properties["generation"] = generation
 
+    def __lt__(self, other):
+        if not isinstance(other, Blob):
+            return super(Blob, self).__lt__(other)
+
+        if self.bucket != other.bucket:
+            raise TypeError("'<' not supported between blobs from different buckets")
```

Comment:
Hmm, I wonder if it would make sense to allow blobs from different buckets to be sorted, using the bucket name?  E.g.:

```python
        return self.bucket.name < other.bucket.name and self.name < other.name
```